mqtrigger/messageQueue/nats.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2016 The Fission Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package messageQueue
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	ns "github.com/nats-io/go-nats-streaming"
+	nsUtil "github.com/nats-io/nats-streaming-server/util"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/fission/fission"
+)
+
+const (
+	natsClusterID  = "fissionMQTrigger"
+	natsProtocol   = "nats://"
+	natsClientID   = "fission"
+	natsQueueGroup = "fission-messageQueueNatsTrigger"
+)
+
+type (
+	Nats struct {
+		MessageQueueTriggerManager
+		nsConn ns.Conn
+	}
+)
+
+func makeNatsTriggerManager(mqTriggerMgr MessageQueueTriggerManager) (MessageQueueTriggerManagerInterface, error) {
```

Comment:
This should not return a trigger manager interface, but rather a new interface, let's call it `MessageQueue`.

The `MessageQueue` interface would just have simple `subscribe(trigger)` and `unsubscribe(trigger)` methods.  Nats can then satisfy that interface.